File: Assignment_2/Q1b.py
def main():
    points = []
    setPoints(points)
    print(points)

    print(" Average distance between points is : ", averageDistance(points),"\n")


def setPoints(points):
    for i in range(1,7):
        for j in range(1,7):
            point = [i,j]
            points.append(point)


def averageDistance(points):
    count = 0.0
    for point_set in points:
        for point_take in points:
            count+= (abs(point_set[0]-point_take[0])+abs(point_set[1]-point_take[1]))
    return count/(36*36)


if __name__ == "__main__":
    main()

File: Assignment_2/haskell/Q1b.hs

main::IO()
main = print $ averageDistance setPoints

setPoints :: [(Double,Double)]
setPoints = [(x,y)|x<-[1..6],y<-[1..6]]

averageDistance :: [(Double,Double)] -> Double
averageDistance nums = sum [abs(x1-x2)+abs(y1-y2)| (x1,y1) <- nums, (x2,y2) <- nums] / (36*36)


